okhttp/src/main/java/okhttp3/CertificatePinner.kt

```diff
@@ -138,29 +138,31 @@ class CertificatePinner internal constructor(
    */
   @Throws(SSLPeerUnverifiedException::class)
   fun check(hostname: String, peerCertificates: List<Certificate>) {
-    var peerCertificates = peerCertificates
+    return check(hostname) {
+      (certificateChainCleaner?.clean(peerCertificates, hostname) ?: peerCertificates)
+          .map { it as X509Certificate }
+    }
+  }
+
+  internal fun check(hostname: String, cleanedPeerCertificatesFn: () -> List<X509Certificate>) {
```

Comment:
I think passing a lambda here is too surprising; it might be better to make this take `cleanedCertificates`, and then make the one callsite guard the call with like `hasPins()` check

Reply:
It's not a simple check without cost if there is pinning in place.  e.g. a client configures pinning on their API host, but not on the CDN (unknown hosts).

In this case, `findMatchingPins(hostname)` is the check we need to do externally rather than hasPins.

Reply:
It's also internal API, for a class with public API.

Reply:
SGTM